src/SDKs/Azure.ApplicationModel.Configuration/data-plane/Azure.ApplicationModel.Configuration/ConfigurationSetting.cs

```diff
@@ -72,20 +73,8 @@ public ConfigurationSetting(string key, string value, string label = null)
         /// A dictionary of tags that can help identify what a key-value may be applicable for.
         /// </summary>
         public IDictionary<string, string> Tags {
-            get {
-                if (_tags == null) {
-                    lock (_key) {
```

Comment:
Model types are not required to be thread-safe but locking on string instances is not the best thing to do in any case. In addition, while Tags getter might be made thread safe using the lock `Dictionary<T,T>` is not threadsafe and concurrent access may cause its data structure corruption and lead to infinite loops.

Reply:
For what it's worth,  +1 for removing the lock.   If we really need to make this thread safe for instantiation, my vote goes to using `Lazy<T>` over a lock.